src/dialects/oracle/index.js

```diff
@@ -179,16 +179,22 @@ assign(Client_Oracle.prototype, {
       default:
         return response;
     }
-  }
+  },
 
+  // If the error is any of these, we'll assume we need to
+  // mark the connection as failed
+  isConnectionError(err) {
```

Comment:
See https://github.com/oracle/odpi/blob/master/src/dpiError.c#L65 for a bigger error list used by the node-oracledb connection pool.  Node-oracledb also uses an OCI call at https://github.com/oracle/odpi/blob/29968fc2085941c40fce8f56a42c9920305d2e29/src/dpiConn.c#L204 which does most of the same checks anyway.